src/components/Panels/Guider/Diagram/Diagram.tsx

```diff
@@ -239,9 +249,14 @@ function Flow() {
 
     setNodes([...sourceNodes, ...initialNodes]);
     setEdges([...sourceEdges, ...initialEdges]);
-    // setTimeout(() => fitView(), 100)
   }, [state]);
 
+  useEffect(() => {
+    if (nodesInitialized) {
+      fitView();
```

Comment:
Oh, that's weird.
Right, the idea of this change is to fit the diagram every time a node is changed. Since we are using the setNodes and setEdges methods provided by the useReactFlow(), I have no idea if they has been already set to the new state.
Exactly as you mentioned, fitView property only works on the initialization.
Let me try now in my desktop machine to see if it works, I just tested in my laptop.

Reply:
I resized my window to make the diagram fairly small. The effect only runs once, after `nodesInitialized` is true and never again (unless `nodesInitialized` is set to false for some reason?)